src/FileAdder/FileAdder.php

```diff
@@ -348,7 +352,7 @@ protected function attachMedia(Media $media)
             return;
         }
 
-        $this->processMediaItem($this->subject, $media, $this);
+        return $this->processMediaItem($this->subject, $media, $this);
```

Comment:
The clearing of the media collection should go in process item. Otherwise the media collection could get cleared before a new item is added.

Reply:
Didn't understand this, can you explain a bit more on this?